### PR TITLE
Milestone Testing

### PR DESCRIPTION
**Pull request by bronnie313**
## Math Magicians Testing 
- Did Setup for React with the React testing library
- Wrote unit tests for operate.js and calculate.js 
- Wrote unit tests for all React components using jest snapshots

### CODE REVIEW

File: src/components/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import calculate from './calculate';
+// import operate from "./operate";
```
Comment (Ghiftee) [inline src/components/logic/calculate.test.js:2]: - [ ] Kindly delete commented code for a cleaner codebase

File: src/components/logic/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  });
+
+  it('should update the "next" value when the buttonName is a number', () => {
+    expect(calculate({ total: '5', next: '2', operation: '+' }, '9')).toEqual({ total: '5', next: '29', operation: '+' });
+  });
+
+  it('should add a decimal point to the "next" value', () => {
+    expect(calculate({ total: '4', next: '6', operation: null }, '.')).toEqual({ total: '4', next: '6.', operation: null });
+  });
+});
```
Comment (Ghiftee) [inline src/components/logic/calculate.test.js:16]: - [ ] Kindly add more tests to cover all the functionalities & scenarios of `calculate.js`

File: src/components/__Test__/Calculator.test.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Cal from '../Calculator';
+
+describe('Calculator component', () => {
+  it('renders correctly', () => {
+    const tree = renderer.create(<Cal />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (Ghiftee) [inline src/components/__Test__/Calculator.test.js:10]: - [ ] Kindly ensure that you use react testing library to simulate user interactions as it is a part of the requirements. To make use of the React Testing Library, you would need to import and utilize its functions such as render, screen, and fireEvent to interact with and test the rendered components.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @bronnie313 and @Paul-tes 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ Usage of Jest snapshots for testing all the components.
✔️ Modules are tested
✔️ Tests pass

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

- [ ] Kinldy update the commands to run tests in your README. The current command there is not correct. Also, kindly fill out the deployment section as it would make your README more professional
Screenshot 2023-06-08 at 16 11 22 (2)


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- I suggest that you add the name of the second author to the README as you both worked on the project together.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by AdnanOlarmmi — APPROVED** [review]
Hi @bronnie313,

This is Adnan Is-haq, your beloved code reviewer, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Approved office man

## Highlights
- Fixed reviewer's changes.
- Tests are passing.

## Optional suggestions
_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**